### ISSUE TITLE
Enable search to be unfiltered

### ISSUE THREAD

**Issue by brent-hartwig** [project-lux/lux-marklogic#223]
**Problem Description:**
There's a good chance LUX can change from filtered searches to unfiltered searches without unacceptable false positive search results (results that the filtering process would have removed).  Benefits:

1. Search results will match the other unfiltered contexts: estimates and facets.  If there is any disparity between those three, it is highly likely a bug in our code (e.g., #37).
2. Unfiltered searches are faster --significantly at times, including with deep pagination (https://github.com/project-lux/lux-marklogic/issues/136#issuecomment-2103398285).
3. Switching to unfiltered search results aligns with our ambitions to switch to Optic, which is unfiltered.

**Expected Behavior/Solution:**
Endpoint consumers may specify whether to filter a search or not.  A default will be provided.  Initially, the default is to be filtered.  The default is to be controlled by a new build property, enabling us to specify the default when deploying the backend's code.

When unfiltered, searches will become faster and may include results due to the following unfiltered search (indexing) limitations.  Should we find other results that a filtered search excludes, we will need to investigate.  We may find an indexing change is needed or that there is an additional limitation / nuance.  Index changes are beyond the scope of this ticket.

Known unfiltered search limitations:

1. Unfiltered searches cannot be whitespace-sensitive.
2. Unfiltered searches cannot be punctuation-sensitive.
3. Unfiltered searches cannot be case-sensitive when the word or phrase is all lowercase.

Examples of index changes we may need to make:

1. Unfiltered searches can only be diacritic-sensitive when the “Fast Diacritic Sensitive Searches” index is enabled.
2. Unfiltered searches can only exclude false positives from phrase and near queries so long as the associated word position index(es) is enabled.

**Requirements:**
- [x] Implement a parameter on the `search` endpoint to specify whether to filter the search results _--also implemented on the facets and relatedList endpoints._
- [x] Document the new `search` endpoint parameter _--also implemented on the facets and relatedList endpoints._
- [x] Implement a new build property to specify the default filtering behavior for the `search` endpoint _--also implemented on the facets and relatedList endpoints._
- [x] Ensure the `searchWillMatch` endpoint remains filtered even the default behavior is unfiltered.
- [x] Add the new build property to the environment-specific properties file in the private repo.

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

~- [ ] Wireframe/Mockup - Mike~
- [x] Committee discussions - Sarah
- [x] Feasibility/Team discussion - Sarah
- [x] Frontend requirements- None required by this ticket although it enables the frontend to implement a filtering option.
- [x] Middle tier requirements - This ticket _might_ require the middle tier deploy updated generated data service proxies.  If/when the frontend pursues a filtering option, the middle tier will need to support the new parameter (multiple endpoints).
- [x] Backend requirements - See above
- [ ] Are new regression tests required for QA - Amy
- [x] Compels a performance test?  Not really.  We should be able to manually compare filtered and unfiltered search results.
- [ ] Questions
- List of questions for discussions. Answers should be documented within the issue.

**UAT/LUX Examples:**
**TODO:** it would be nice to compile a list of searches that draw out differences between filtered and unfiltered search results.

**Dependencies/Blocks:**
This ticket neither blocks nor is blocked by another ticket.  Once implemented, it will enable the frontend and middle tier to implement a filtering option.

**Related Github Issues:**
Links included above.

**Related links:**
None

**Wireframe/Mockup:**
N/A

**Comment by brent-hartwig** [comment]
Decided on 12 Jul with @jffcamp, @prowns, @azaroth42 to implement this but that search results are to remain filtered by default until a mid-July dataset is in both DEV and TST (or two DEV tenants?).

**Comment by brent-hartwig** [comment]
Started work in the [223-unfiltered-cts-search branch](https://github.com/project-lux/lux-marklogic/tree/223-unfiltered-cts-search).

**Comment by brent-hartwig** [comment]
Implemented in PR https://github.com/project-lux/lux-marklogic/pull/231, which may end up in release1.21.

**Comment by brent-hartwig** [comment]
On 1 Aug 24, Peter enabled unfiltered searches in DEV's main tenant.

**Comment by jffcamp** [comment]
Decision made to not test. Not able to identify tests where a difference can be surfaced. ML only applies filters when needed. So, we will allow filtering to remain on. Dev will be updated to re-enable filtering.

**Comment by brent-hartwig** [comment]
@jffcamp, were you looking for a difference in results (not ideal) or better performance (the objective)?

**Comment by roamye** [comment]
IT Team Meeting 08/16: 
It was discussed in [today's meeting](https://docs.google.com/document/d/1j3PIcYl4ZfLqlVJX0ErPXNT781vFMVtMvYxot_eysfA/edit) that further investigation is needed before making a decision. Such as understanding the impact and how to test.

**Comment by roamye** [comment]
IT Team Meeting [11/05](https://docs.google.com/document/d/1om0op78Tqe5ZI0DFZRmQRvtEb7TJ8wddqfd3a9Z50AA/edit?tab=t.0): 
This issue is reopening as the team wants to make searches **unfiltered by default** to...
- make it easier when switching to OPTIC 
- confirm/test the idea

Next steps: 
- make searches unfiltered by default. 
- develop a testing process 
- identify tickets that need filtering
- run tests by end of Nov/early Dec

@brent-hartwig @clarkepeterf for the testing processes how many simple searches are needed from site improve?

**Comment by brent-hartwig** [comment]
> for the testing processes how many simple searches are needed from site improve?

@roamye, is the goal to try to surface some searches that give different results when filtered vs. not filtered?  Note that simple search is limited to a single search pattern.  Also note we have a script that serve up the JSON-formatted search criteria from production logs.

cc: @clarkepeterf

**Comment by brent-hartwig** [comment]
@clarkepeterf, I committed cleaned up versions of ours scripts within the [scripts/estimateVersusActual](https://github.com/project-lux/lux-marklogic/tree/223-compare-filtered-to-unfiltered/scripts/estimateVersusActual) directory of the 223-[compare-filtered-to-unfiltered](https://github.com/project-lux/lux-marklogic/tree/223-compare-filtered-to-unfiltered) branch.  Directions are in the first script.  There is a repeated TODO in the second, should we choose to take it on: "Gracefully handle requests that time out."

**Comment by xinjianguosccs** [comment]
@brent-hartwig Jeff asked me to "compare filtered and unfiltered search result counts of production searches (ML 223)"

Where and how should I start with? Thanks

**Comment by xinjianguo** [comment]
@brent-hartwig  Jeff asked me to "compare filtered and unfiltered search result counts of production searches (ML 223)"

Where and how should I start with? Thanks

**Comment by brent-hartwig** [comment]
Hi, @xinjianguo,

Crack open [step-01-mineSearchCriteria.sh](https://github.com/project-lux/lux-marklogic/blob/main/scripts/estimateVersusActual/step-01-mineSearchCriteria.sh) and review the opening comment and go by the directions therein.

For step no. 3, it's probably easier to write your own scp command to get (all 30) *8004-ErrorLog*.txt files from Blue and Green (versus modifying [collectBackendLogs.sh](https://github.com/project-lux/lux-marklogic/blob/main/scripts/logAnalysis/collectBackendLogs.sh)).  Optionally, you can isolate the production logs from the test logs.   I'm of the opinion not to bother and expect the process to take longer.

After you're all set with the input files, continue with the next steps.  

There are two "TODO: Gracefully handle requests that time out." comments in the companion script, (step-02-createSearchReport.sh)[https://github.com/project-lux/lux-marklogic/blob/main/scripts/estimateVersusActual/step-02-createSearchReport.sh].  I don't know if you'll encounter timeouts.  If you do, it may be sufficient to temporarily increase the timeout settings (max and default) on the application server with port 8004 in the selected environment.  As for the selected environment, there's no need to use PROD.

Holler if you have any questions!

**Comment by xinjianguo** [comment]
@brent-hartwig one run took more than one day, the result file is attached to  https://git.yale.edu/lux-its/ml-cluster-formation/issues/78. Let me know if more runs are needed.

**Comment by brent-hartwig** [comment]
For a given blue/green environment, we would ideally want two two-week stretches when the ML cluster was part of production.
